modules/sharethroughBidAdapter.js

```diff
@@ -1,123 +1,178 @@
+
+    const nonHttp = sharethroughInternal.getProtocol().indexOf('http') < 0;
+    const secure = nonHttp || (sharethroughInternal.getProtocol().indexOf('https') > -1);
+
+    const req = {
+      id: utils.generateUUID(),
+      at: 1,
+      cur: ['USD'],
+      tmax: timeout,
+      site: {
+        domain: window.location.hostname,
+        page: window.location.href,
+        ref: bidderRequest.refererInfo ? bidderRequest.refererInfo.referer || null : null,
+      },
+      user: {
+        ext: {
+          eids: userIdAsEids(bidRequests[0]),
+        },
+      },
+      device: {
+        ua: navigator.userAgent,
+        language: navigator.language,
+        js: 1,
+        dnt: navigator.doNotTrack === '1' ? 1 : 0,
+        h: window.screen.height,
+        w: window.screen.width,
+      },
+      regs: {
+        coppa: config.getConfig('coppa') === true ? 1 : 0,
+        ext: {},
+      },
+      source: {
+        ext: {
+          version: '$prebid.version$',
+          str: VERSION,
+          schain: bidRequests[0].schain,
+        },
+      },
+      bcat: bidRequests[0].params.bcat || [],
+      badv: bidRequests[0].params.badv || [],
+      test: 0,
+    };
+
+    if (bidderRequest.gdprConsent) {
+      const gdprApplies = bidderRequest.gdprConsent.gdprApplies === true;
+      req.regs.ext.gdpr = gdprApplies ? 1 : 0;
+      if (gdprApplies) {
+        req.user.ext.consent = bidderRequest.gdprConsent.consentString;
       }
+    }
 
-      if (bidRequest.schain) {
-        query.schain = JSON.stringify(bidRequest.schain);
-      }
+    if (bidderRequest.uspConsent) {
+      req.regs.ext.us_privacy = bidderRequest.uspConsent;
+    }
 
-      const floor = getFloor(bidRequest);
-      if (floor) {
-        query.bidfloor = floor;
-      }
+    const imps = bidRequests.map(bidReq => {
+      const impression = {};
 
-      if (bidRequest.params.badv) {
-        query.badv = bidRequest.params.badv;
+      const gpid = utils.deepAccess(bidReq, 'ortb2Imp.ext.data.pbadslot');
+      if (gpid) {
+        impression.ext = { gpid: gpid };
       }
 
-      if (bidRequest.params.bcat) {
-        query.bcat = bidRequest.params.bcat;
+      if (bidReq.mediaTypes && bidReq.mediaTypes.video) {
+        const videoRequest = bidReq.mediaTypes.video;
+        impression.video = {
+          pos: nullish(videoRequest.pos, 0),
+          topframe: utils.inIframe() ? 0 : 1,
+          skip: nullish(videoRequest.skip, 0),
+          linearity: nullish(videoRequest.linearity, 1),
+          minduration: nullish(videoRequest.minduration, 5),
+          maxduration: nullish(videoRequest.maxduration, 60),
+          playbackmethod: videoRequest.playbackmethod || [2],
+          api: getVideoApi(videoRequest),
+          mimes: videoRequest.mimes || ['video/mp4'],
+          protocols: getVideoProtocols(videoRequest),
+          w: videoRequest.playerSize[0],
+          h: videoRequest.playerSize[1],
```

Comment:
May help to verify playerSize array is included in your `isBidRequestValid` function for video requests since there is no default that can be set.

Reply:
good catch - I set some defaults